webtau-core/src/main/java/com/twosigma/webtau/expectation/code/ThrowExceptionMatcher.java

```diff
@@ -98,4 +97,15 @@ public boolean matches(CodeBlock codeBlock) {
 
         return isEqual;
     }
+
+    private void extractExceptionDetails(Throwable t) {
+        if (t instanceof UndeclaredThrowableException) {
+            Throwable undeclaredCheckedException = t.getCause();
```

Comment:
Could cause ever be null?

Reply:
Not if the proxy does its job correctly, I think. The docs for the exception say 

> As of release 1.4, this exception has been retrofitted to conform to the general purpose exception-chaining mechanism. The "undeclared checked exception that was thrown by the invocation handler" that may be provided at construction time and accessed via the getUndeclaredThrowable() method is now known as the cause, and may be accessed via the Throwable.getCause() method, as well as the aforementioned "legacy method."